### PR TITLE
feature/revert site settings

### PR DESCRIPTION
**Pull request by AndriyAndriyovuch**
## Code reviewers

- [ ] @loqimean 

## Summary of issue

The site settings should provide a way for the admin user to revert to the default favicon and title if desired. (#513)

## Summary of change

- added revert button
- added locales to button
- changed min width to 110px for all buttons

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

Знімок екрану з 2023-09-16 17-35-40

### CODE REVIEW

File: app/views/account/site_settings/edit.html.slim [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+        .col-12.has-float-label.my-auto
+          = f.input :title, class: "form-control", input_html: { data: { action: "input->handle-browser-tab#setTitle" } }
+          = f.input :favicon, class: "form-control", input_html: { data: {  action: "change->handle-browser-tab#setIcon" } }
+          = render "browser_tab", site_setting: @site_setting
 
-    = f.submit t("buttons.update")
-      i.fa.fa-times-circle
+      = f.submit t("buttons.update"), class: "inline"
+
+    = button_to t("buttons.revert"), account_site_settings_revert_path, onclick: "return confirm('#{j(t('.confirm_default'))}')", class: "btn-revert position-absolute right-bottom"
```
Comment (loqimean) [inline app/views/account/site_settings/edit.html.slim:14]: що це за конфірм тут?) хібе що `data: { turbo_confirm: "some text" }`

File: config/routes.rb [hunk 2]
```diff
@@ -43,6 +43,7 @@
       resource :app_config, only: [:edit, :update]
       patch "/feature_flags", to: "feature_flags#update", as: "features_flags"
       resource :site_setting, only: [:edit, :update]
+      post "site_settings/revert", to: "site_settings#revert"
```
Comment (loqimean) [inline config/routes.rb:46]: Таке має спрацювати
```suggestion
      resource :site_setting, only: [:edit, :update] do
        post :revert
      end
```

File: spec/requests/site_setting_spec.rb [hunk 3]
```diff
@@ -52,4 +52,29 @@
...
+
+    context "with valid params" do
+      let(:site_setting) { SiteSetting.current }
+      let(:site_setting_params) { FactoryBot.attributes_for(:site_setting, :custom_setting) }
+      let(:site_setting_default_params) { FactoryBot.attributes_for(:site_setting, :with_valid_site_setting) }
+
+      before { site_setting.update(site_setting_params) }
+
+      it "reverts site setting" do
+        get account_site_settings_revert_path
```
[the remark this answers is not in the document]
Comment (AndriyAndriyovuch) [reply spec/requests/site_setting_spec.rb]: метод не той прописав, все гуд

File: app/controllers/account/site_settings_controller.rb [hunk 4]
```diff
@@ -17,6 +17,24 @@ def update
...
+      @site_setting.update(title: "ZeroWaste", favicon: {
+        io: File.open("app/assets/images/logo_zerowaste.png"),
+        filename: "logo_zerowaste.png",
+        content_type: "image/png"
+      })
+
+      flash[:notice] = t("notifications.site_setting_reverted")
+    rescue
+      flash[:alert] = t("notifications.site_setting_not_reverted")
+    end
```
Comment (loqimean) [inline app/controllers/account/site_settings_controller.rb:33]: а навіщо тут бегін-рескʼю?)

File: app/controllers/account/site_settings_controller.rb [hunk 5]
```diff
@@ -17,6 +17,18 @@ def update
     end
   end
 
+  def revert
+    @site_setting = resource
+
+    @site_setting.update(title: "ZeroWaste", favicon: {
```
[the remark this answers is not in the document]
Comment (loqimean) [reply app/controllers/account/site_settings_controller.rb:23]: де?))

**Review by loqimean — APPROVED** [review]

**Pull request merged**